api/base/serializers.py

```diff
@@ -147,6 +169,64 @@ def to_representation(self, value):
         return {'links': {self.link_type: {'href': url, 'meta': meta}}}
 
 
+class JSONAPIHyperlinkedRelatedField(ser.HyperlinkedRelatedField):
+    """
+    HyperlinkedRelated field that returns a nested dict with url,
+    optional meta information, and link_type.
+
+    Example:
+
+        branched_from = JSONAPIHyperlinkedRelatedField(view_name='nodes:node-detail', lookup_field='pk',
+                                                    lookup_url_kwarg='node_id', read_only=True, link_type='related')
+
+    """
+    def __init__(self, view_name=None, **kwargs):
+        self.json_api_link = True  # serializes to a links object
+        self.meta = kwargs.pop('meta', {})
+        self.link_type = kwargs.pop('link_type', 'url')
+        super(JSONAPIHyperlinkedRelatedField, self).__init__(view_name=view_name, **kwargs)
+
+    def to_representation(self, value):
+        """
+        Returns nested dictionary in format {'links': {'self.link_type': ... }
+
+        If no meta information, self.link_type is equal to a string containing link's URL.  Otherwise,
+        the link is represented as a links object with 'href' and 'meta' members.
+        """
+        url = super(JSONAPIHyperlinkedRelatedField, self).to_representation(value)
+        meta = _rapply(self.meta, _url_val, obj=value, serializer=self.parent)
+        return {'links': {self.link_type: {'href': url, 'meta': meta}}}
+
+
+class JSONAPIHyperlinkedGuidRelatedField(ser.Field):
+    """
+    Field that returns a nested dict with the url (constructed based
+    on the object's type), optional meta information, and link_type.
+
+    Example:
+
+        target = JSONAPIHyperlinkedGuidRelatedField(link_type='related', meta={'type': 'get_target_type'})
+
+    """
+    def __init__(self, **kwargs):
+        self.json_api_link = True  # serializes to a links object
```

Comment:
This should be a class attribute rather than an instance attribute, since it will not vary from instance to instance.